src/quacc/recipes/newtonnet/ts.py

```diff
@@ -293,3 +307,250 @@ def _get_hessian(atoms: Atoms) -> NDArray:
+    nimages (int): Number of images. Default is 17.
+    sweep (bool): Sweep across the path optimizing one image at a time.
+                  Default is to perform sweeping updates if there are more than 35 atoms.
+    output (str): Output filename. Default is "interpolated.xyz".
+    tol (float): Convergence tolerance. Default is 2e-3.
+    maxiter (int): Maximum number of minimization iterations. Default is 15.
+    microiter (int): Maximum number of micro iterations for sweeping algorithm. Default is 20.
+    scaling (float): Exponential parameter for Morse potential. Default is 1.7.
+    friction (float): Size of friction term used to prevent very large change of geometry. Default is 1e-2.
+    dist_cutoff (float): Cut-off value for the distance between a pair of atoms to be included in the coordinate system. Default is 3.
+    save_raw (str): When specified, save the raw path after bisections but before smoothing. Default is None.
+    """
+    # Read the initial geometries.
+    symbols = r_p_atoms[0].get_chemical_symbols()
+
+    X = [conf.get_positions() for conf in r_p_atoms]
+
+    if len(X) < 2:
+        raise ValueError("Need at least two initial geometries.")
+
+    # First redistribute number of images. Perform interpolation if too few and subsampling if too many images are given
+    raw = redistribute(symbols, X, nimages, tol=tol * 5)
+    if save_raw is not None:
+        write_xyz(save_raw, symbols, raw)
+
+    # Perform smoothing by minimizing distance in Cartesian coordinates with redundant internal metric
+    # to find the appropriate geodesic curve on the hyperspace.
+    smoother = Geodesic(symbols, raw, scaling, threshold=dist_cutoff, friction=friction)
+    if sweep is None:
+        sweep = len(symbols) > 35
+    try:
+        if sweep:
+            smoother.sweep(tol=tol, max_iter=maxiter, micro_iter=microiter)
+        else:
+            smoother.smooth(tol=tol, max_iter=maxiter)
+    finally:
+        # Save the smoothed path to output file. try block is to ensure output is saved if one ^C the process, or there is an error
+        write_xyz(output, symbols, smoother.path)
+    return symbols, smoother.path
+
+
+def setup_images(logdir: str, xyz_r_p: str, n_intermediate: int = 40):
+    """
+    Sets up intermediate images for NEB calculations between reactant and product states.
+
+    Parameters:
+    logdir (str): Directory to save the intermediate files.
+    xyz_r_p (str): Path to the XYZ file containing reactant and product structures.
+    n_intermediate (int): Number of intermediate images to generate.
+
+    Returns:
+    List: List of ASE Atoms objects with calculated energies and forces.
+    """
+    calc_defaults = {
+        "model_path": SETTINGS.NEWTONNET_MODEL_PATH,
+        "settings_path": SETTINGS.NEWTONNET_CONFIG_PATH,
+    }
+
+    calc_flags = recursive_dict_merge(calc_defaults, {})
+
+    try:
+        # Ensure the log directory exists
+        os.makedirs(logdir, exist_ok=True)
```

Comment:
This is not going to be possible because we cannot be writing out to custom directories.